Visualization/CreateBlankResults.py

```diff
@@ -5,88 +5,151 @@
-filesToReplace = sys.argv[1]
-geneList = sys.argv[2]
-
-# File extension check
-if not filesToReplace.endswith(".txt"):
-    print("[Warning] " + geneList + " may not be a text file!")
-
-# File extension check
-if not geneList.endswith(".faa"):
-    print("[Warning] " + geneList + " may not be a FASTA file!")
-
-# Reads sequence file list and stores it as a string.
-try:
-    with open(filesToReplace, "rU") as newFile:
-        filesToReplace = newFile.read()
-        newFile.close()
-except IOError:
-    print("Failed to open " + filesToReplace)
-    exit(1)
-
-# Splits string into a list. Each element is a single line from the string.
-filesToReplaceList = filesToReplace.splitlines()
-
-FakeResults = []
-try:
-    handle = open(geneList, "rU")
-    SeqRecords = SeqIO.parse(handle, "fasta")
-    for record in SeqRecords:
-        FakeResults.append(
-            record.id + ",sseqid,0,evalue,qcovhsp,bitscore")  # qseqid sseqid pident evalue qcovhsp bitscore
-    handle.close()
-except IOError:
-    print("Failed to open " + filesToReplace + " or " + geneList)
-    exit(1)
-
-FakeResultsOut = "\n".join(FakeResults)
-
-print(">> Writing Fake Results...")
-
-for x in filesToReplaceList:
-    replacementFileName = x + ".csv"
+# Gives a timestamp (string) for help with logging
+def ts():
+    current_time = datetime.datetime.utcnow()
+    time_formatted = current_time.strftime("%a %b %d %H:%M:%S UTC %Y")
+
+    timestamp = "[ " + time_formatted + " ]: "
+
+    return(timestamp)
+
+
+# Checks whether or not the provided input_csv is an empty file. Returns logical.
+def check_if_input_CSV_is_empty(input_csv):
+    # Returns True is the CSV is empty and False if not
+
+    if os.stat(input_csv).st_size == 0:
+        return True
+
+    elif os.stat(input_csv).st_size > 0:
+        return False
+
+    else:
+        print(ts() + "ERROR: the input file appears to neither be empty nor non-empty. Exiting...")
+        exit(1)
+
+
+# Checks whether or not the provided file has the expected extension. Throws warning if not. No return.
+def file_extension_check(filename, extension):
+    # Throws a warning if file extension does not appear to be appropriate
+
+    if extension == "csv":
+        if not filename.endswith(".csv"):
+            print(ts() + "WARNING: '" + filename + "' may not be a CSV file!")
+
+    elif extension == "faa":
+        if not filename.endswith(".faa"):
+            print(ts() + "WARNING: '" + filename + "' may not be an amino acid FASTA file!")
+
+    else:
+        print(ts() + "ERROR: extension must be 'csv' or 'faa'. Exiting...")
+        exit(1)
+
+
+# Uses input query_proteins FAA file to build a blank BLAST CSV table. Returns the table.
+def generate_blank_results(query_proteins):
+
+    blank_results_list = []
+
+    try:
+        handle = open(query_proteins, "rU")
+        fasta_entries = SeqIO.parse(handle, "fasta")
+        for entry in fasta_entries:
+            blank_results_list.append(
```

Comment:
You could probably use python CSV module for this. However, it's probably not worth importing in this use case.

Reply:
https://docs.python.org/3/library/csv.html

Reply:
So You Want To Write Your Own CSV code?:

http://thomasburette.com/blog/2014/05/25/so-you-want-to-write-your-own-CSV-code/